### PR TITLE
WIP: Fix bazelci

### PR DESCRIPTION
**Pull request by acmcarther**
This (eventually) fixes bazel ci.

### CODE REVIEW

**Comment by acmcarther** [comment]
`cargo_raze_examples` seems seriously broken on windows, so I'll probably punt on fixing that for right now. Focusing on fixing osx and linux now.

**Comment by UebelAndre** [comment]
> `cargo_raze_examples` seems seriously broken on windows, so I'll probably punt on fixing that for right now. Focusing on fixing osx and linux now.

I'd expect windows to be broken. I made a "best guess" when doing the bazel implementation but didn't have a windows machine to test on 😞

**Comment by acmcarther** [comment]
It's still pretty broken, and I don't have any more cycles to work on this today due to IRL stuff. I'll take another look tomorrow morning.

- Most Recent Run: https://buildkite.com/bazel/cargo-raze/builds/19#0b3f4809-c590-4267-b8b3-03c11b8d411b
- Logs: https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/0b3f4809-c590-4267-b8b3-03c11b8d411b/impl/cargo_raze_lib_test/attempt_1.log

Do you have access to this @UebelAndre? If not that should be fixed. Also, a dump of the relevant part of the logs:

```

failures:

---- planning::tests::test_semver_matching stdout ----
thread 'planning::tests::test_semver_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `cargo` from `/tmp/.tmpf2cdUy`")', impl/src/planning.rs:597:74
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- planning::tests::test_subplan_produces_crate_root_with_forward_slash stdout ----
thread 'planning::tests::test_subplan_produces_crate_root_with_forward_slash' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/var/lib/buildkite-agent/.cargo/bin/cargo metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: No such file or directory (os error 2)
    1: No such file or directory (os error 2)', impl/src/planning.rs:164:74

---- checks::tests::test_check_all_vendored_verifies_vendored_state stdout ----
thread 'checks::tests::test_check_all_vendored_verifies_vendored_state' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/var/lib/buildkite-agent/.cargo/bin/cargo metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: No such file or directory (os error 2)
    1: No such file or directory (os error 2)', impl/src/testing.rs:310:54

---- checks::tests::test_check_resolve_matches_packages_fails_correctly stdout ----
thread 'checks::tests::test_check_resolve_matches_packages_fails_correctly' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/var/lib/buildkite-agent/.cargo/bin/cargo metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: No such file or directory (os error 2)
    1: No such file or directory (os error 2)', impl/src/metadata.rs:614:54
```

(among similar)

That path looks approximately right, but the "failed to start `cargo metadata` is pretty suspicious.

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ tasks:
   ubuntu1804:
     shell_commands:
```
Comment (UebelAndre) [inline .bazelci/presubmit.yml:8]: You should no longer need cargo.

Comment (UebelAndre) [reply .bazelci/presubmit.yml:8]: I've gotten a PR through `rules_rust` to support hermetic testing with environment variables.

**Comment by acmcarther** [comment]
Yep, I see that you got this fixed.

(I was able to get the jank version of this that pulls cargo down working _as well_, but clearly making the test hermetic is preferred to my hack)

I'm going to close this and open a separate PR that just disables the windows build. That will put us at green.

**Pull request closed without merging**